.gitlab/source_test/golang_deps_generate.yml

```diff
@@ -0,0 +1,22 @@
+---
+# golang_deps_generate stage
+# Contains the step to build the golang dependency chain and preserve it
+golang_deps_generate:
+  rules:
+    !reference [.on_main_or_release_branch]
+  stage: source_test
+  image: 486234852809.dkr.ecr.us-east-1.amazonaws.com/ci/datadog-agent-buildimages/deb_x64:$DATADOG_AGENT_BUILDIMAGES
+  tags: ["runner:main"]
+  needs: ["linux_x64_go_deps"]
```

Comment:
If this job really needs the `linux_x64_go_deps` output it should do something with it